src/dsp_tools/utils/excel_to_json/excel_to_json_properties.py

```diff
@@ -0,0 +1,470 @@
+    """
+    err_msg_list = [f"The 'properties' section defined in the Excel file '{excel_filename}' did not pass validation."]
+    json_path_to_property = regex.search(r"^\$\[(\d+)\]", validation_error.json_path)
+    if json_path_to_property:
+        # fmt: off
+        wrong_property_name = (
+            jsonpath_ng.ext.parse(json_path_to_property.group(0))
+            .find(list_properties)[0]
+            .value["name"]
+        )
+        # fmt: on
+        excel_row = int(json_path_to_property.group(1)) + 2
+        err_msg_list.append(f"The problematic property is '{wrong_property_name}' in Excel row {excel_row}.")
+        affected_field = regex.search(
+            r"name|labels|comments|super|subject|object|gui_element|gui_attributes",
+            validation_error.json_path,
+        )
+        if affected_field:
+            err_msg_list.append(
+                f"The problem is that the column '{affected_field.group(0)}' has an invalid value: "
+                f"{validation_error.message}"
+            )
+    else:
+        err_msg_list.append(
+            f"The error message is: {validation_error.message}\n" f"The error occurred at {validation_error.json_path}"
+        )
+    return "\n".join(err_msg_list)
+
+
+def _validate_properties(
+    properties_list: list[dict[str, Any]],
+    excelfile: str,
+) -> bool:
+    """
+    This function checks if the "properties" section of a JSON project file is valid, according to the JSON schema,
+    and if the property names are unique.
+
+    Args:
+        properties_list: the "properties" section of a JSON project as a list of dicts
+        excelfile: path to the Excel file containing the properties
+
+    Raises:
+        UserError: if the validation fails
+
+    Returns:
+        True if the "properties" section passed validation
+    """
+    with importlib.resources.files("dsp_tools").joinpath("resources/schema/properties-only.json").open(
+        encoding="utf-8"
+    ) as schema_file:
+        properties_schema = json.load(schema_file)
+    try:
+        jsonschema.validate(instance=properties_list, schema=properties_schema)
+    except jsonschema.ValidationError as err:
+        err_msg = _search_validation_error(
+            list_properties=properties_list, excel_filename=excelfile, validation_error=err
+        )
+        raise UserError(err_msg) from None
+    return True
+
+
+def _search_convert_numbers(value_str: str) -> str | int | float:
+    """
+    This function takes a string and searches if the string contains a float or an integer. In those cases, it converts
+    the string to the corresponding data type. If it is not a float or integer, it returns the string as is.
+
+    Args:
+        value_str: The value which is checked and may be converted
+
+    Returns:
+        A int if the string was an integer, float if the string was a float or str if it was neither
+    """
+    if regex.search(r"^\d+\.\d+$", value_str):
+        return float(value_str)
```

Comment:
this would miss `"1e+3"` which is a valid representation of `1000.0` in python

Reply:
Thanks I did not think of that. I will resolve this issue separately.